Lib/textwrap.py

```diff
@@ -432,15 +432,30 @@ def dedent(text):
     lines = text.split('\n')
 
     # Get length of leading whitespace, inspired by ``os.path.commonprefix()``.
-    non_blank_lines = [l for l in lines if l and not l.isspace()]
-    l1 = min(non_blank_lines, default='')
-    l2 = max(non_blank_lines, default='')
-    margin = 0
+    val = False
+    for i, line in enumerate(lines):
+        # Compute min + max concurrently + normalize others
+        if line and not line.isspace():
+            if val:
+                if line < l1:
+                    l1 = line
+                elif line > l2:
+                    l2 = line
+            else:
+               val = True
+               l1 = l2 = line 
+                
+        else:
+            lines[i] = ''
+    
+    if not val or not l1:
+        return '\n'.join(lines)
+    
```

Comment:
```suggestion
    l1 = 'z'
    l2 = ''
    for i, line in enumerate(lines):
        # Compute min + max concurrently + normalize others
        if line and not line.isspace():
            if line < l1:
                l1 = line
            if line > l2:
                l2 = line
        else:
            lines[i] = ''

    if not l1:
        return '\n'.join(lines)

    margin = 0
    for margin, c in enumerate(l2):
        if c != l1[margin] or c not in ' \t':
            break
    
```
This saves chechking the `val`. Performance gain is only a bit though.

I agree with the other reviewers: unless we find out a way to make this much faster, the small gains are not worth the change.

@Marius-Juston Thanks for putting in the work to get all the performance numbers!